vmm/src/vmm_config/drive.rs

```diff
@@ -158,72 +157,84 @@ impl BlockDeviceConfigs {
     }
 
     fn create(&mut self, block_device_config: BlockDeviceConfig) -> Result<()> {
-        // check if the path exists
+        // If no path exists, then an error is returned.
         if !block_device_config.path_on_host.exists() {
-            return Err(DriveError::InvalidBlockDevicePath);
+            Err(DriveError::InvalidBlockDevicePath)
         }
-
-        if self
+        // If the path already exists, then an error is returned.
+        else if self
             .get_index_of_drive_path(&block_device_config.path_on_host)
             .is_some()
         {
-            return Err(DriveError::BlockDevicePathAlreadyExists);
+            Err(DriveError::BlockDevicePathAlreadyExists)
         }
-
-        // check whether the Device Config belongs to a root device
-        // we need to satisfy the condition by which a VMM can only have on root device
-        if block_device_config.is_root_device {
-            if self.has_root_block {
-                return Err(DriveError::RootBlockDeviceAlreadyAdded);
-            } else {
-                self.has_root_block = true;
-                self.read_only_root = block_device_config.is_read_only;
-                self.has_partuuid_root = block_device_config.partuuid.is_some();
-                // Root Device should be the first in the list whether or not PARTUUID is specified
-                // in order to avoid bugs in case of switching from partuuid boot scenarios to
-                // /dev/vda boot type.
-                self.config_list.push_front(block_device_config);
-            }
-        } else {
+        // If the added device is not a root block device, then it can
+        // be unconditionally added now.
+        else if !block_device_config.is_root_device {
             self.config_list.push_back(block_device_config);
+            Ok(())
+        }
+        // Otherwise, if there already is a root block device, an error
+        // is returned.
+        else if self.has_root_block {
+            Err(DriveError::RootBlockDeviceAlreadyAdded)
+        }
+        // Otherwise, configure a root block device.
+        else {
+            self.has_root_block = true;
+            self.read_only_root = block_device_config.is_read_only;
+            self.has_partuuid_root = block_device_config.partuuid.is_some();
+            // Root Device should be the first in the list whether or not PARTUUID is specified
+            // in order to avoid bugs in case of switching from partuuid boot scenarios to
+            // /dev/vda boot type.
+            self.config_list.push_front(block_device_config);
+            Ok(())
         }
-
-        Ok(())
     }
 
     /// Updates a Block Device Config. The update fails if it would result in two
     /// root block devices.
     fn update(&mut self, mut index: usize, new_config: BlockDeviceConfig) -> Result<()> {
-        // Check if the path exists
+        // If no path exists, then an error is returned.
```

Comment:
Same as before (comments should explain the why and I don't think you need the else-if).